custom_components/cover_time_based/cover.py

```diff
@@ -362,90 +401,311 @@ async def async_set_cover_tilt_position(self, **kwargs):
     async def async_close_cover(self, **kwargs):
         """Turn the device close."""
         _LOGGER.debug("async_close_cover")
-        current_position = self.travel_calc.current_position()
-        if current_position is None or current_position < 100:
+        
+        current_travel_position = self.travel_calc.current_position()
+        if current_travel_position is None or current_travel_position < 100:
+            # Cancel any active delay task (we're starting new movement)
+            self._cancel_delay_task()
+            
+            # Calculate how much travel will move (to fully closed = 100)
+            travel_distance = 100 - (current_travel_position if current_travel_position is not None else 0)
+            # Calculate time this movement will take
+            movement_time = (travel_distance / 100.0) * self._travel_time_down
+            
             self.travel_calc.start_travel_down()
             self.start_auto_updater()
-            self._update_tilt_before_travel(SERVICE_CLOSE_COVER)
+            
+            # Tilt will change during travel - calculate proportional movement
+            if self._has_tilt_support():
+                # Calculate how much tilt should move in the same time
+                tilt_distance = (movement_time / self._tilting_time_down) * 100.0
+                
+                # Get current tilt position and calculate new position (moving to vertical = 100)
+                current_tilt_position = self.tilt_calc.current_position()
+                new_tilt_position = min(100, current_tilt_position + tilt_distance)

```

Comment:
comments stating obvious

```suggestion
                tilt_distance = (movement_time / self._tilting_time_down) * 100.0
                
                current_tilt_position = self.tilt_calc.current_position()
                new_tilt_position = min(100, current_tilt_position + tilt_distance)
```